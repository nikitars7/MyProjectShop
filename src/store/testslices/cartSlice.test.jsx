import cartSliceReducer, {
  addToCart,
  removeFromCart,
} from "../slices/cartSlice";
describe("cartSlice", () => {
  test("should return default state", () => {
    const result = cartSliceReducer(undefined, { type: "" });
    expect(result).toEqual({ cartItems: [], totalPrice: 0 });
    expect(result).toMatchSnapshot();
  });
  test("should add a new item to the state", () => {
    const action = {
      type: addToCart.type,
      payload: {
        id: 1,
        imageUrl: "image",
        name: "added",
        price: 100,
        count: 1,
        size: "L",
      },
    };
    const result = cartSliceReducer({ cartItems: [], totalPrice: 0 }, action);
    const { cartItems, totalPrice } = result;
    expect(cartItems[0].name).toBe("added");
    expect(totalPrice).toBe(100);
    expect(result).toMatchSnapshot();
  });
  test("should remove the item from the state", () => {
    const cartItem = {
      id: 1,
      imageUrl: "image",
      name: "added",
      price: 100,
      count: 1,
      size: "L",
    };
    const action = {
      type: removeFromCart.type,
      payload: 1,
    };
    const result = cartSliceReducer(
      { cartItems: [cartItem], totalPrice: 100 },
      action
    );
    expect(result).toEqual({ cartItems: [], totalPrice: 0 });
    expect(result).toMatchSnapshot();
  });
});
