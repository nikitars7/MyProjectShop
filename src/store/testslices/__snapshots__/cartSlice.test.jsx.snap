// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cartSlice should add a new item to the state 1`] = `
Object {
  "cartItems": Array [
    Object {
      "count": 1,
      "id": 1,
      "imageUrl": "image",
      "name": "added",
      "price": 100,
      "size": "L",
    },
  ],
  "totalPrice": 100,
}
`;

exports[`cartSlice should remove the item from the state 1`] = `
Object {
  "cartItems": Array [],
  "totalPrice": 0,
}
`;

exports[`cartSlice should return default state 1`] = `
Object {
  "cartItems": Array [],
  "totalPrice": 0,
}
`;
